### PR TITLE
chore: devcontainer

### PR DESCRIPTION
**Pull request by evilru**
add a devcontainer to make contribution easier

### CODE REVIEW

**Comment by marianfoo** [comment]
@evilru great idea!
it´s docker, so i guess its usable in Linux and Windows right?

**Comment by evilru** [comment]
Yes, it works for Windows,  macOS and Linux.

**Comment by vobu** [comment]
i'll review this along with #52 wrt to: will use the devcontainer, check out #52 as doc source and validate that things work™

File: .devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+This will
+
+* Clone the Repository in a Container Volume
+* Build the Docker Image
+* Start the Docker Container and map the required ports
+* Mount the created Container Volume
+* Install the required npm packages
+* Install the required ruby gems
+
+The only thing left to do is open the console in Visual Studio Code (it is attached to the running Development Container), navigate into the docs folder and start the Development Server as described in the [README.md](../README.md).
```
Comment (vobu) [inline .devcontainer/README.md:25]: might be worth providing a shortcut/brief summary here à la
```sh
$> cd docs
$> bundle exec jekyll serve --livereload
```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		"dbaeumer.vscode-eslint",
+    "editorconfig.editorconfig",
+    "davidanson.vscode-markdownlint"
+	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	"forwardPorts": [4000, 35729],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "npm install && sudo gem install bundler && cd docs && bundle install",
```
Comment (vobu) [inline .devcontainer/devcontainer.json:25]: probably even append `&& bundle exec jekyll serve --livereload`?
if so, needs mention in the `.devcontainer/README.md` that a manual jekyll start is not required

**Review by vobu — COMMENTED** [review]
this is so dope! 🤙 
worked right out of the box.
THANKS A TON! 🙌 
i'd suggest two things:
- add the `.devcontainer/README.md` content to the `/README.md`
- either auto-start jekyll in postcreate or give the user a hint in the README how to proceed after container start/build is done

**Comment by vobu** [comment]
ah, one more thing :)
what do you think of pre-installing the `gh cli` in the dev container?
that would make it easier to switch between github PRs in the devcontainer and have them "previewed" live via jekyll
i'll prepare the commands as a comment in the dockerfile

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.183.0/containers/javascript-node/.devcontainer/base.Dockerfile
+
+# [Choice] Node.js version: 16, 14, 12
+ARG VARIANT="16-buster"
+FROM mcr.microsoft.com/vscode/devcontainers/javascript-node:0-${VARIANT}
+
+# [Optional] Uncomment this section to install additional OS packages.
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends ruby-full
+
```
Comment (vobu) [inline .devcontainer/Dockerfile:10]:
```shell
# taken straight from https://github.com/cli/cli/blob/trunk/docs/install_linux.md
curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo gpg --dearmor -o /usr/share/keyrings/githubcli-archive-keyring.gpg 
echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
sudo apt update
sudo apt install gh
```

Comment (marianfoo) [reply .devcontainer/Dockerfile:10]: great idea, few lines how to start would be great to
probably just use [https://cli.github.com/manual/gh_pr_checkout](https://cli.github.com/manual/gh_pr_checkout)

**Comment by marianfoo** [comment]
worked great on my private pc!
although the company proxy will probably argue against it

**Review by treee111 — APPROVED** [review]
works like a charm - setup in < 10 minutes! great job 👍

for reference: after the pull i had to
- open terminal
- `cd docs/`
- `bundle exec jekyll serve --livereload`
- open http://localhost:4000/UI5-Best-Practice/

I think it is a good idea to integrate the review-comments!

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (evilru) [reply .devcontainer/Dockerfile:13]: I added the gh cli to the container

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (evilru) [reply .devcontainer/devcontainer.json]: unfortunately it's not that easy :(

* the postCreateCommand runs just once during the first time setup
* vscode overwrites the entrypoint of the container to have a running process during the setup
   see overrideCommand at https://code.visualstudio.com/docs/remote/devcontainerjson-reference 
* set overrideCommand to false and add bundle exec jekyll serve --livereload as ENTRYPOINT in the dockerfile doesn't work either as the postCreateCommand did not run yet.. :(

File: .devcontainer/README.md (hunk 1 above)
Comment (evilru) [reply .devcontainer/README.md]: added to the readme

**Comment by evilru** [comment]
I added gh cli as suggested and moved the readme content into the project readme.
From my perspective everything is resolved.

@vobu, if you don't have more suggestions - could you please merge the pr?
I am not sure about the workflow, especially if I am allowed to merge my own pr...🙂

**Review by vobu — APPROVED** [review]
again, great stuff 👏

**Pull request merged**